### PR TITLE
Andrei/agora 2162 delegate filtering

### PR DESCRIPTION
**Pull request by andreitr**
Displaying only allow-listed delegates

---

## PR-Codex overview
This PR focuses on removing `Tenant.current()` calls and implementing an allow-list filter for delegate lists.

### Detailed summary
- Removed `Tenant.current()` calls from `DelegateCardList.tsx`
- Implemented allow-list filtering for delegate lists in `getDelegates.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/app/api/common/delegates/getDelegates.ts [hunk 1]
```diff
@@ -135,7 +135,62 @@ async function getDelegates({
...
+            FROM ${namespace + ".delegates"}
+            WHERE delegate = ANY($1)
+            ORDER BY voting_power DESC
+            OFFSET $2
+            LIMIT $3;
+            `,
+          allowList,
+          skip,
+          take
+        );
```
Comment (jefag) [inline src/app/api/common/delegates/getDelegates.ts:191]: Would it be better to give this default case an explicit label, and error if a sort method wasn't passed?  Forces the caller to be explicit.

**Review by jefag — APPROVED** [review]

File: src/app/api/common/delegates/getDelegates.ts [hunk 2]
```diff
@@ -135,7 +135,62 @@ async function getDelegates({
...
+
+  // Applies allow-list filtering to the delegate list
+  const paginatedAllowlistQuery = async (skip: number, take: number) => {
+    switch (sort) {
+      case "most_delegators":
+        return prisma.$queryRawUnsafe<DelegatesGetPayload[]>(
+          `
+            SELECT *
+            FROM ${namespace + ".delegates"}
+            WHERE num_of_delegators IS NOT NULl AND delegate = ANY($1)
```
Comment (mcgingras) [inline src/app/api/common/delegates/getDelegates.ts:151]: does this `NULl` spacing issue matter at all

Comment (andreitr) [reply src/app/api/common/delegates/getDelegates.ts:151]: good catch, strange...

File: src/app/api/common/delegates/getDelegates.ts (hunk 1 above)
Comment (andreitr) [reply src/app/api/common/delegates/getDelegates.ts:191]: @jefag can you give me an example?

File: src/app/api/common/delegates/getDelegates.ts [hunk 3]
```diff
@@ -135,7 +135,62 @@ async function getDelegates({
...
+        return prisma.$queryRawUnsafe<DelegatesGetPayload[]>(
+          `
+            SELECT *
+            FROM ${namespace + ".delegates"}
+            WHERE voting_power > 0 AND delegate = ANY($2)
+            ORDER BY -log(random()) / voting_power
+            OFFSET $3
+            LIMIT $4;
+            `,
+          seed,
```
Comment (mcgingras) [inline src/app/api/common/delegates/getDelegates.ts:172]: maybe we don't need seed in this one

Comment (andreitr) [reply src/app/api/common/delegates/getDelegates.ts:172]: @mcgingras that list is randomized and the seed ensures that results do not repeat.

Comment (mcgingras) [reply src/app/api/common/delegates/getDelegates.ts:172]: oh I was just thinking that I didn't see $1 used anywhere in the query

**Pull request merged**